src/pudl/scripts/dbt_helper.py

```diff
@@ -201,6 +201,76 @@ def to_yaml(self, schema_path: Path):
             yaml_output = _prettier_yaml_dumps(self.model_dump(exclude_none=True))
             schema_file.write(yaml_output)
 
+    def merge_metadata_from(self, old_schema: "DbtSchema") -> "DbtSchema":
+        """Merge metadata from an old schema into this one, preferring new values where present."""
+
+        def merge_tables(new_tables, old_tables):
+            old_table_map = {t.name: t for t in old_tables or []}
+            merged = []
+
+            for new_table in new_tables or []:
+                old_table = old_table_map.get(new_table.name)
+                if not old_table:
+                    merged.append(new_table)
+                    continue
+
+                old_columns = {c.name: c for c in old_table.columns or []}
+                merged_columns = [
+                    (
+                        col.model_copy(
+                            update={
+                                "description": col.description
+                                or old_columns.get(col.name, {}).description,
+                                "data_tests": col.data_tests
+                                or old_columns.get(col.name, {}).data_tests,
+                                "tags": col.tags or old_columns.get(col.name, {}).tags,
+                            }
+                        )
+                        if col.name in old_columns
+                        else col
+                    )
+                    for col in new_table.columns or []
+                ]
+
+                merged_table = new_table.model_copy(
+                    update={
+                        "description": new_table.description or old_table.description,
+                        "data_tests": new_table.data_tests or old_table.data_tests,
+                        "tags": new_table.tags or old_table.tags,
+                        "meta": new_table.meta or old_table.meta,
+                        "columns": merged_columns,
```

Comment:
I'm a little nervous about the specific enumeration of all these properties. Do we know we've covered all the things we can ever encounter? If we were to manually add a new field that's compatible with dbt but not with our `DbtSchema` or `DbtModel` etc. Pydantic classes, then we'd get a validation error when reading in the existing schema and have to fix things, right? So it wouldn't happen silently and accidentally remove something.

Reply:
Yes, if we add properties directly to the schema.yml and try to load a `DbtSchema.from_yaml` from that schema, it should throw a ValidationError. However, this only protects us from adding details to the schemas that are not implemented in the classes. We would still need to have a way to ensure any new info added to the classes are also added to the method that transfers them over. I will see if there is some elegant way to ensure all metadata is preserved.